fix: lazy compilation request handler should filter lazy-compilation request (#9166)
web-infra-dev/rspack: website/docs/en/config/experiments.mdx

The current lazy compilation aligns with the webpack implementation, **and is still in the experimental stage**. In some scenarios, lazy compilation might not work as expected, or the performance improvement may be insignificant.
:::

## experiments.lazyCompilation.backend

- **Type:** `LazyCompilationBackendOptions`

```ts

> type LazyCompilationBackendOptions = {

Reviewer: We can change L114 from `backend?: { ... }` to `LazyCompilationBackendOptions` to reduce duplication of types in the document

  experiments: {
    lazyCompilation: {
      backend: {
        server() {
          return yourServer;
        },
      },
    },
  },
};
```

You can provide a custom server instance to handle requests from clients for `lazy-compilation`. The provided server must meet the following interface requirements.

```ts
import type { IncomingMessage, ServerResponse } from 'node:http';
import type { AddressInfo, ListenOptions, Socket } from 'node:net';

interface Server {
  // Register callback function to handle requests
  on(
    event: 'request',
    callback: (req: IncomingMessage, res: ServerResponse) => void,
  ): void;

  // Register callback function for new client connections
  on(event: 'connection', callback: (socket: Socket) => void): void;

  // Register callback function for new client connections
  on(event: 'listening', callback: (err?: Error) => void): void;

  // Unregister registered callbacks
  off(event: string, callback: () => void): void;

  // Show address information of the server
  address(): AddressInfo;

  close(callback: (err?: any) => void): void;

  listen(listenOption: number | ListenOption): void;
}
```

### lazyCompilation.backend.listen

> - **Type:** `number|ListenOption`

Reviewer: ```suggestion
- **Type:** `number | ListenOptions | ((server: Server) => void)`
```

```ts
type ListenOption = {
  port?: number | undefined;
  host?: string | undefined;
  backlog?: number | undefined;
  path?: string | undefined;
  exclusive?: boolean | undefined;
  readableAll?: boolean | null;
  writableAll?: boolean | null;
  ip6Only?: boolean | null;
};
```

> Specify how the sever listens ,you can pass in object form listening configuration,such as specifying listening ports,you can also pass in functions control how listen is done .

Reviewer: ```suggestion
Specify how the sever listens ,you can pass in object form listening configuration,such as specifying listening ports,you can also pass in functions control how listen is done.
```

> If you provide your own service instance,the service's listen is called by provider unless explicitly configured with functional forms of listening.

Reviewer: ```suggestion
When providing a custom server instance, you are responsible for calling the server's `listen` method to start the service, unless you explicitly configure a function forms` listen` option.
```
